code/modules/vehicles/motorbike.dm

```diff
@@ -0,0 +1,137 @@
+#define FUEL_PER_CAN_POUR 50
+
+/obj/vehicle/ridden/motorbike
+	name = "all-terrain motorbike"
+	desc = "An all-terrain vehicle built for traversing rough terrain with ease. \"TGMC CAVALRY\" is stamped on the side of the engine."
+	icon_state = "motorbike"
+	max_integrity = 300
+	soft_armor = list("melee" = 30, "bullet" = 30, "laser" = 30, "energy" = 0, "bomb" = 30, "fire" = 60, "acid" = 60)
+	obj_flags = CAN_BE_HIT
+	resistance_flags = XENO_DAMAGEABLE
+	key_type = null
+	integrity_failure = 0.5
+	buckle_flags = CAN_BUCKLE|BUCKLE_PREVENTS_PULL|BUCKLE_NEEDS_HAND
+	///Internal motorbick storage object
+	var/obj/item/storage/internal/motorbike_pack/motor_pack = /obj/item/storage/internal/motorbike_pack
+	///Mutable appearance overlay that covers up the mob with th e bike as needed
+	var/mutable_appearance/motorbike_cover
+	///Fuel count, fuel usage is one per tile moved
+	var/fuel_count = 0
+	///max fuel that this bike can hold
+	var/fuel_max = 400
+	COOLDOWN_DECLARE(enginesound_cooldown)
+
+/obj/vehicle/ridden/motorbike/Initialize()
+	. = ..()
+	AddElement(/datum/element/ridable, /datum/component/riding/vehicle/motorbike)
+	motor_pack = new motor_pack(src)
+	motorbike_cover = mutable_appearance(icon, "motorbike_cover", MOB_LAYER + 0.1)
+	fuel_count = fuel_max
+
+/obj/vehicle/ridden/motorbike/examine(mob/user)
+	. = ..()
+	if(!ishuman(user))
+		return
+	to_chat(user, "To access internal storage click with an empty hand or drag the bike onto self.")
+	to_chat(user, "The fuel gauge on the bike reads \"[fuel_count/fuel_max*100]%\"")
+
+/obj/vehicle/ridden/motorbike/post_buckle_mob(mob/living/M)
+	add_overlay(motorbike_cover)
+	return ..()
+
+/obj/vehicle/ridden/motorbike/post_unbuckle_mob(mob/living/M)
+	if(!LAZYLEN(buckled_mobs))
+		cut_overlay(motorbike_cover)
+	return ..()
+
+/obj/vehicle/ridden/motorbike/welder_act(mob/living/user, obj/item/I)
+	if(obj_integrity >= max_integrity)
+		return TRUE
+	balloon_alert_to_viewers("[user] starts repairs", ignored_mobs = user)
+	balloon_alert(user, "You start repair")
+	if(!do_after(user, 2 SECONDS))
```

Comment:
is the do_after preventing from spamming?

Reply:
Yes